### PR TITLE
Copy files under src to destination by module type

### PR DESCRIPTION
**Pull request by gilgardosh**
Makes root files be copied to dist, and ./src files copied to esm/cjs/both

### CODE REVIEW

**Comment by gilgardosh** [comment]
I removed [this export](https://github.com/kamilkisiela/bob/pull/173/commits/b33a07d22f3176016cc8661d5e54ce47ec5fd7d0) from the test, since it's false (the file path is `./src/styles.css`.

But I didn't address case of
```
"exports": {
  ...
  "./src/style.css": "./src/style.css"
}
```
(Currently the export will be copied as-is)

I think it can be merged, with the fix for in-code relative paths, and this edge case addressed later.

File: src/commands/build.ts [hunk 1]
```diff
@@ -466,16 +466,40 @@ export function validatePackageJson(
...
+          }
+
+          // FOr each type, copy files to the relevant directory
+          const allTypesExec: Array<Promise<void>> = [];
+          allTypes.forEach(type => {
+            allTypesExec.push(
+              executeCopy(sourcePath, join(distDir, file.replace('src/', `${type}/`))),
+            );
+          });
+          await Promise.all(allTypesExec);
```
Comment (B2o5T) [inline src/commands/build.ts:498]:
```suggestion
await Promise.all(
  allTypes.map(type =>
    executeCopy(sourcePath, join(distDir, file.replace('src/', `${type}/`)))
  )
)
```

File: src/commands/build.ts [hunk 2]
```diff
@@ -466,16 +466,40 @@ export function validatePackageJson(
...
   return Promise.all(
     allFiles.map(async file => {
       if (await fse.pathExists(join(cwd, file))) {
         const sourcePath = join(cwd, file);
-        const destPath = join(distDir, file.replace('src/', ''));
-        await mkdirp(dirname(destPath));
-        await fse.copyFile(sourcePath, destPath);
+        if (file.includes('src/')) {
+          // Figure relevant module types
+          const allTypes: string[] = [];
```
Comment (B2o5T) [inline src/commands/build.ts:483]:
```suggestion
          const allTypes: ('cjs' | 'esm')[] = [];
```

Comment (n1ru4l) [reply src/commands/build.ts]:
```suggestion
          const allTypes: Array<'cjs' | 'esm'> = [];
```

**Review by n1ru4l — APPROVED** [review]

**Pull request merged**